Working auto generation

dpeter99 opened the pull request:
First pass on this.

So the basic idea is that the ``out`` folder should never be commited. We should be able to regenerate everything it has from the other files. So it got removed and gitignored.
Doxygen has a config file format so everything got added to that. The logo is embeded using the default doxygen parameter for it. Later we can add custom header for the page throug doxygen.
The new folders are
- ``java`` this folder is where you will put the original forge sources and never commit them as well. (The only file in the folder that get's uploaded is the README.txt to make git create the folder.)
- ``src_doc`` this folder is where all the actual doc files live. They don't have to be in the same directory structure as the java sources but we should still keep the folders.
- ``src_web`` this is the folder for the sataic site resources (custom images, templates etc.)

dpeter99 committed:
Working auto generation

TheCurle commented on Doxyfile:
```diff
@@ -0,0 +1,2613 @@
+# entered, it will be relative to the location where doxygen was started. If
+# left blank the current directory will be used.
+
+OUTPUT_DIRECTORY       = "P:\Minecraft\Forge Dev\Forge_Community_Wiki\javadocs\out"
```
Alas, the problem with Doxygen.
My P drive is a remote fileserver.
We need to think of a better solution than this.

TheCurle commented on Doxyfile:
```diff
@@ -0,0 +1,2613 @@
+# qualified scopes will look different, etc.
+# The default value is: NO.
+
+OPTIMIZE_OUTPUT_JAVA   = NO
```
Why is this not optimised for Java?

TheCurle commented on Doxyfile:
This will cause issues if we add the MC sources as a (hidden) folder.

TheCurle commented on Doxyfile:
More hardcoded nonexistent paths

TheCurle commented on Doxyfile:
Why so many files? We work exclusively with .java and .dox classes.

TheCurle replied:
And .json.. and .toml

TheCurle commented on Doxyfile:
Set this to yes

TheCurle reviewed (changes requested):
![image](https://user-images.githubusercontent.com/42079760/97354797-d81a7500-188d-11eb-91c6-17365c551311.png)

TheCurle reviewed (approved):
Hmmmmmmmmm.

dpeter99 committed:
CLI Test

dpeter99 committed:
No Doxy warnings

dpeter99 committed:
Forge is patched now

dpeter99 committed:
Added extra gradle setup call

dpeter99 committed:
Hide undoc classes

sciwhiz12 committed:
Auto-generate documentation using Doxygen; deploy to GitHub pages

Move Doxygen generation to GH Action instead of Gradle

Do not generate sources when publishing

Simplify GH publish action to three steps

Add advancement.dox test file

Uppercased filename

Use dark theme

Use .java instead of .dox

Use Gradle in GH action for gen. docs

Revert to using Doxygen action

Attempt using doxygen commands in .java files

Re-add Gradle to GH action, add incremental build support to tasks

Rename files

Auto-generate documentation using Doxygen; deploy to GitHub pages

Fix ordering of GH action steps

Document new method, for testing caching

dpeter99 committed:
Deep clone and version picker gui

dpeter99 committed:
Updated forge to 1.16.3-34.1.35

dpeter99 committed:
Action file touch up

dpeter99 committed:
change java folder

dpeter99 committed:
Versioning

dpeter99 committed:
Version selector

dpeter99 committed:
Fix my crappy re-base.

This was my first try using it so don't hate me.

dpeter99 committed:
Fixes and removed unneeded files

sciwhiz12 commented on .github/workflows/build_docs.yml:
```diff
@@ -0,0 +1,122 @@
+      # - name: Gradle Get version
+      #   uses: eskatos/gradle-command-action@v1
+      #   with:
+      #     arguments: getVersion -q
```
As this is commented out, can this part be removed?

sciwhiz12 commented on .github/workflows/build_docs.yml:
```diff
@@ -0,0 +1,122 @@
+          key: ${{ steps.forge_version.outputs.prop }}-sources
+
+      # - name: Generate and copy sources using Gradle
+      #   if: steps.cache.outputs.cache-hit != 'true'
```
Same here: as this is commented out, could this be removed? Especially since it looks to be a copy of the next step.

sciwhiz12 commented on .github/workflows/build_docs.yml:
```diff
@@ -0,0 +1,122 @@
+          target: "out/html/version.js"
+
+      # - name: Generate and copy sources using Gradle
+      #   uses: eskatos/gradle-command-action@v1
```
Same here: as it's commented out, can this be removed?

sciwhiz12 commented on src/web/version.js:
```diff
@@ -0,0 +1,27 @@
+
+}
+
+
```
Maybe remove some of the whitespace here.

sciwhiz12 commented on src/web/forge_style.css:
```diff
@@ -0,0 +1,8 @@
+}
+
+#titlearea {
+    padding-left: 10px;
```
Instead of adding padding (which causes the horizontal scrollbar to appear), could we add a custom id to the version picker and apply a margin to that element instead?

TheCurle reviewed (approved):
Not got the time to test it.
I dig the custom CSS though.

Gradle tasks look good, and i've seen the results in the FCW discord so i'm happy with this

sciwhiz12 reviewed (changes requested):
Just a few more tweaks, then this will be ready for merging _(finally)_.

dpeter99 committed:
Cleanup

dpeter99 committed:
Build for versioned branches as well
